### PR TITLE
Support Vitis-AI quantization for onnx models in Olive

### PR DESCRIPTION
**Pull request by sheng-xiao**
## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`

### CODE REVIEW

**Comment by sheng-xiao** [comment]
@microsoft-github-policy-service agree company="AMD"

File: olive/vitis_ai/calibrate.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python
```
Comment (trajepl) [inline olive/vitis_ai/calibrate.py:1]: Maybe we can just keep one copyright declaration.

File: examples/vitis_ai/retinaface_ptq_cpu/user_script.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+from olive.constants import Framework
+from olive.evaluator.accuracy import AccuracyScore
+from olive.model import OliveModel
+
+
+class ResNetDataReader(CalibrationDataReader):
+
+    def __init__(self, calibration_image_folder, augmented_model_path):
+        self.image_folder = calibration_image_folder
+        self.augmented_model_path = augmented_model_path
```
Comment (trajepl) [inline examples/vitis_ai/retinaface_ptq_cpu/user_script.py:25]: How about to accept model path in under the component of vitis_ai?
In some cases, vitis quantization could be after conversion where the model path is generated dynamically. For examples:
1. conversion -> optimization -> vistis quantization -> ....
In this optimization chain, user may not know the model path, then it is hard to set it in olive config.

File: examples/vitis_ai/resnet_ptq_cpu/resnet_config.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{
+    "verbose": true,
+    "input_model":{
+        "type": "ONNXModel",
+        "config": {
+            "model_path": "models/resnet18.onnx",
+            "is_file": true
```
Comment (jambayk) [inline examples/vitis_ai/resnet_ptq_cpu/resnet_config.json:7]: `is_file` is not supported anymore. It has been changed to `"model_storage_kind": "file"`

**Review by devang-ml — COMMENTED** [review]
- olive/vitis_ai folder content should be in the same folder where vitis quantization pass is located.
- If the vitis quantization pass is operating on an ONNX model then it should reside in olive/passes/onnx folder.

**Comment by guotuofeng** [comment]
remember to run pre-commit to format the code.

File: olive/passes/vitis_ai/__init__.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (C) 2023, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (devang-ml) [inline olive/passes/vitis_ai/__init__.py:2]: Do you still need __init__.py ?

Comment (sheng-xiao) [reply olive/passes/vitis_ai/__init__.py]: No, files in /psses/vitis_ai/ has been moved to /passes/onnx, this file is OK to remove. Thanks very much for your review!

**Review by trajepl — DISMISSED** [review]

File: examples/test_resnet_vitis_ai_ptq_cpu.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    olive_config["engine"]["target"] = system
+
+    if system == "aml_system":
+        generate_olive_workspace_config("olive-workspace-config.json")
+
+    footprint = olive_run(olive_config)
+    check_output(footprint)
+
+
+def generate_olive_workspace_config(workspace_config_path):
```
Comment (trajepl) [inline examples/test_resnet_vitis_ai_ptq_cpu.py:68]: Please refer to the case in main branch to update way for workspace config generation.

File: examples/resnet_vitis_ai_ptq_cpu/resnet_static_config.json [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                }
+            }
+        }
+    },
+    "systems": {
+        "local_system": {"type": "LocalSystem"},
+        "aml_system": {
+            "type": "AzureML",
+            "config": {
+                "aml_config_path": "olive-workspace-config.json",
```
Comment (trajepl) [inline examples/resnet_vitis_ai_ptq_cpu/resnet_static_config.json:24]: Please remove the `aml_config_path` then update it with following config:

https://github.com/microsoft/Olive/blob/0d9c1f7260698659902452c8500e80823641da3b/examples/resnet_ptq_cpu/resnet_static_config.json#L3

**Review by trajepl — APPROVED** [review]

**Review by guotuofeng — APPROVED** [review]

**Review by trajepl — APPROVED** [review]

**Pull request merged**